In [1]:
import scanpy as sc
import pandas as pd

In [2]:
from os import listdir
from os.path import isfile, join

In [3]:
outputPath="/home/lugli/spuccio/Projects/SP039/FileH5AD_input/"

In [4]:
def h5ad_create(folderID,outputPath):
    folderPath = "/home/lugli/spuccio/Projects/SP035_TregIntegration/Glioma_Locati_Lugli/"
    tmp1 = "adata"+folderID
    tmp1 = sc.read(folderPath+folderID+"/outs/filtered_feature_bc_matrix/matrix.mtx.gz")
    tmp1 = tmp1.transpose()
    genes = pd.read_csv(folderPath+ folderID + "/outs/filtered_feature_bc_matrix/features.tsv.gz",header=None,sep="\t")
    barcodes = pd.read_csv(folderPath+ folderID +"/outs/filtered_feature_bc_matrix/barcodes.tsv.gz",header=None,sep="\t")
    genes = genes[[0,1]]
    genes =genes.rename({0: 'ENSGeneID', 1: 'GeneSymbol'}, axis=1)
    tmp1.var = genes
    tmp1.var.index = genes['GeneSymbol']
    del tmp1.var['GeneSymbol']
    barcodes =barcodes.rename({0: 'barcode'}, axis=1)
    tmp1.obs = barcodes
    tmp1.obs.index = barcodes['barcode']
    del tmp1.obs['barcode']
    tmp1.var_names_make_unique()
    print(tmp1)
    tmp1.write_h5ad(outputPath+folderID+'.h5ad')

In [5]:
onlydir = [f for f in listdir("/home/lugli/spuccio/Projects/SP035_TregIntegration/Glioma_Locati_Lugli/") if not isfile(join("/home/lugli/spuccio/Projects/SP035_TregIntegration/Glioma_Locati_Lugli/", f))]

In [6]:
for _ in range(len(onlydir)):
    print(onlydir[_])
    h5ad_create(onlydir[_],outputPath)

S21227_Lib_BS_06
AnnData object with n_obs × n_vars = 4985 × 33538
    var: 'ENSGeneID'
S21230_Lib_BS_09
AnnData object with n_obs × n_vars = 3014 × 33538
    var: 'ENSGeneID'
S21226_Lib_BS_05
AnnData object with n_obs × n_vars = 6741 × 33538
    var: 'ENSGeneID'
S21232_Lib_BS_T-10
AnnData object with n_obs × n_vars = 4530 × 33538
    var: 'ENSGeneID'
S21228_Lib_BS_07
AnnData object with n_obs × n_vars = 9212 × 33538
    var: 'ENSGeneID'
S21231_Lib_BS_S-10
AnnData object with n_obs × n_vars = 4032 × 33538
    var: 'ENSGeneID'
S21229_Lib_BS_08
AnnData object with n_obs × n_vars = 4553 × 33538
    var: 'ENSGeneID'
S21225_Lib_BS_03
AnnData object with n_obs × n_vars = 1557 × 33538
    var: 'ENSGeneID'
S21233_Lib_BS_S11
AnnData object with n_obs × n_vars = 6401 × 33694
    var: 'ENSGeneID'


In [7]:
GLIO1_T = sc.read("/home/lugli/spuccio/Projects/SP039/FileH5AD_input/"+"S21225_Lib_BS_03"+".h5ad")
GLIO2_T = sc.read("/home/lugli/spuccio/Projects/SP039/FileH5AD_input/"+"S21226_Lib_BS_05"+".h5ad")
GLIO3_T = sc.read("/home/lugli/spuccio/Projects/SP039/FileH5AD_input/"+"S21227_Lib_BS_06"+".h5ad")
GLIO4_T = sc.read("/home/lugli/spuccio/Projects/SP039/FileH5AD_input/"+"S21228_Lib_BS_07"+".h5ad")
GLIO5_T = sc.read("/home/lugli/spuccio/Projects/SP039/FileH5AD_input/"+"S21229_Lib_BS_08"+".h5ad")
GLIO6_T = sc.read("/home/lugli/spuccio/Projects/SP039/FileH5AD_input/"+"S21230_Lib_BS_09"+".h5ad")
GLIO7_H = sc.read("/home/lugli/spuccio/Projects/SP039/FileH5AD_input/"+"S21231_Lib_BS_S-10"+".h5ad")
GLIO7_T = sc.read("/home/lugli/spuccio/Projects/SP039/FileH5AD_input/"+"S21232_Lib_BS_T-10"+".h5ad")
GLIO8_H = sc.read("/home/lugli/spuccio/Projects/SP039/FileH5AD_input/"+"S21233_Lib_BS_S11"+".h5ad")

In [8]:
GLIO1_T.var['GeneTemp']= list(GLIO1_T.var.index)
GLIO2_T.var['GeneTemp']= list(GLIO2_T.var.index)
GLIO3_T.var['GeneTemp']= list(GLIO3_T.var.index)
GLIO4_T.var['GeneTemp']= list(GLIO4_T.var.index)
GLIO5_T.var['GeneTemp']= list(GLIO5_T.var.index)
GLIO6_T.var['GeneTemp']= list(GLIO6_T.var.index)
GLIO7_H.var['GeneTemp']= list(GLIO7_H.var.index)
GLIO7_T.var['GeneTemp']= list(GLIO7_T.var.index)
GLIO8_H.var['GeneTemp']= list(GLIO8_H.var.index)

In [9]:
GLIO1_T.var['GeneTempClean']= list(GLIO1_T.var['GeneTemp'].str.split(".",0, expand=True)[0])
GLIO2_T.var['GeneTempClean']= list(GLIO2_T.var['GeneTemp'].str.split(".",0, expand=True)[0])
GLIO3_T.var['GeneTempClean']= list(GLIO3_T.var['GeneTemp'].str.split(".",0, expand=True)[0])
GLIO4_T.var['GeneTempClean']= list(GLIO4_T.var['GeneTemp'].str.split(".",0, expand=True)[0])
GLIO5_T.var['GeneTempClean']= list(GLIO5_T.var['GeneTemp'].str.split(".",0, expand=True)[0])
GLIO6_T.var['GeneTempClean']= list(GLIO6_T.var['GeneTemp'].str.split(".",0, expand=True)[0])
GLIO7_H.var['GeneTempClean']= list(GLIO7_H.var['GeneTemp'].str.split(".",0, expand=True)[0])
GLIO7_T.var['GeneTempClean']= list(GLIO7_T.var['GeneTemp'].str.split(".",0, expand=True)[0])
GLIO8_H.var['GeneTempClean']= list(GLIO8_H.var['GeneTemp'].str.split(".",0, expand=True)[0])

In [10]:
GLIO1_T.var.index = GLIO1_T.var['GeneTempClean']
GLIO2_T.var.index = GLIO2_T.var['GeneTempClean']
GLIO3_T.var.index = GLIO3_T.var['GeneTempClean']
GLIO4_T.var.index = GLIO4_T.var['GeneTempClean']
GLIO5_T.var.index = GLIO5_T.var['GeneTempClean']
GLIO6_T.var.index = GLIO6_T.var['GeneTempClean']
GLIO7_H.var.index = GLIO7_H.var['GeneTempClean']
GLIO7_T.var.index = GLIO7_T.var['GeneTempClean']
GLIO8_H.var.index = GLIO8_H.var['GeneTempClean']

In [11]:
GLIO1_T.var_names_make_unique()
GLIO2_T.var_names_make_unique()
GLIO3_T.var_names_make_unique()
GLIO4_T.var_names_make_unique()
GLIO5_T.var_names_make_unique()
GLIO6_T.var_names_make_unique()
GLIO7_H.var_names_make_unique()
GLIO7_T.var_names_make_unique()
GLIO8_H.var_names_make_unique()

In [12]:
del GLIO1_T.var['GeneTemp']
del GLIO1_T.var['GeneTempClean']

del GLIO2_T.var['GeneTemp']
del GLIO2_T.var['GeneTempClean']

del GLIO3_T.var['GeneTemp']
del GLIO3_T.var['GeneTempClean']

del GLIO4_T.var['GeneTemp']
del GLIO4_T.var['GeneTempClean']

del GLIO5_T.var['GeneTemp']
del GLIO5_T.var['GeneTempClean']

del GLIO6_T.var['GeneTemp']
del GLIO6_T.var['GeneTempClean']

del GLIO7_H.var['GeneTemp']
del GLIO7_H.var['GeneTempClean']

del GLIO8_H.var['GeneTemp']
del GLIO8_H.var['GeneTempClean']


In [13]:
GLIO1_T.obs['SampleID'] = "GLIO1_T"
GLIO1_T.obs['GEO_RNA'] = "NotAvailable"
GLIO1_T.obs['Cohort'] = "NotAvailable"
GLIO1_T.obs['Type'] = "PrimaryTumor"  
GLIO1_T.obs['Subtype'] = "NotAvailable"
GLIO1_T.obs['Diagnosis'] = "Glioblastoma"
GLIO1_T.obs['Grade'] = "NotAvailable"
GLIO1_T.obs['Location'] = "NotAvailable"
GLIO1_T.obs['IDH1_Status'] = "WT"
GLIO1_T.obs['EGFR_Status'] = "NotAvailable"
GLIO1_T.obs['MGMT'] = "Met"
GLIO1_T.obs['CellCount'] = "1557"
GLIO1_T.obs['Sample'] = "Patient29"
GLIO1_T.obs['Sex'] = "F"
GLIO1_T.obs['Age'] = "NotAvailable"
GLIO1_T.obs['Sorted'] = "CD45"

In [14]:
GLIO2_T.obs['SampleID'] = "GLIO2_T"
GLIO2_T.obs['GEO_RNA'] = "NotAvailable"
GLIO2_T.obs['Cohort'] = "NotAvailable"
GLIO2_T.obs['Type'] = "PrimaryTumor"  
GLIO2_T.obs['Subtype'] = "NotAvailable"
GLIO2_T.obs['Diagnosis'] = "Astrocytoma"
GLIO2_T.obs['Grade'] = "NotAvailable"
GLIO2_T.obs['Location'] = "NotAvailable"
GLIO2_T.obs['IDH1_Status'] = "Mut"
GLIO2_T.obs['EGFR_Status'] = "NotAvailable"
GLIO2_T.obs['MGMT'] = "Met"
GLIO2_T.obs['CellCount'] = "6741"
GLIO2_T.obs['Sample'] = "Patient30"
GLIO2_T.obs['Sex'] = "F"
GLIO2_T.obs['Age'] = "NotAvailable"
GLIO2_T.obs['Sorted'] = "CD45"

In [15]:
GLIO3_T.obs['SampleID'] = "GLIO3_T"
GLIO3_T.obs['GEO_RNA'] = "NotAvailable"
GLIO3_T.obs['Cohort'] = "NotAvailable"
GLIO3_T.obs['Type'] = "PrimaryTumor"  
GLIO3_T.obs['Subtype'] = "NotAvailable"
GLIO3_T.obs['Diagnosis'] = "Glioblastoma"
GLIO3_T.obs['Grade'] = "NotAvailable"
GLIO3_T.obs['Location'] = "NotAvailable"
GLIO3_T.obs['IDH1_Status'] = "WT"
GLIO3_T.obs['EGFR_Status'] = "NotAvailable"
GLIO3_T.obs['MGMT'] = "UnMet"
GLIO3_T.obs['CellCount'] = "4985"
GLIO3_T.obs['Sample'] = "Patient31"
GLIO3_T.obs['Sex'] = "M"
GLIO3_T.obs['Age'] = "NotAvailable"
GLIO3_T.obs['Sorted'] = "CD45"

In [16]:
GLIO4_T.obs['SampleID'] = "GLIO4_T"
GLIO4_T.obs['GEO_RNA'] = "NotAvailable"
GLIO4_T.obs['Cohort'] = "NotAvailable"
GLIO4_T.obs['Type'] = "PrimaryTumor"  
GLIO4_T.obs['Subtype'] = "NotAvailable"
GLIO4_T.obs['Diagnosis'] = "Glioblastoma"
GLIO4_T.obs['Grade'] = "NotAvailable"
GLIO4_T.obs['Location'] = "NotAvailable"
GLIO4_T.obs['IDH1_Status'] = "WT"
GLIO4_T.obs['EGFR_Status'] = "NotAvailable"
GLIO4_T.obs['MGMT'] = "Met"
GLIO4_T.obs['CellCount'] = "9212"
GLIO4_T.obs['Sample'] = "Patient32"
GLIO4_T.obs['Sex'] = "M"
GLIO4_T.obs['Age'] = "NotAvailable"
GLIO4_T.obs['Sorted'] = "CD45"

In [17]:
GLIO5_T.obs['SampleID'] = "GLIO5_T"
GLIO5_T.obs['GEO_RNA'] = "NotAvailable"
GLIO5_T.obs['Cohort'] = "NotAvailable"
GLIO5_T.obs['Type'] = "PrimaryTumor"  
GLIO5_T.obs['Subtype'] = "NotAvailable"
GLIO5_T.obs['Diagnosis'] = "Astrocytoma"
GLIO5_T.obs['Grade'] = "NotAvailable"
GLIO5_T.obs['Location'] = "NotAvailable"
GLIO5_T.obs['IDH1_Status'] = "Mut"
GLIO5_T.obs['EGFR_Status'] = "NotAvailable"
GLIO5_T.obs['MGMT'] = "Met"
GLIO5_T.obs['CellCount'] = "4553"
GLIO5_T.obs['Sample'] = "Patient33"
GLIO5_T.obs['Sex'] = "F"
GLIO5_T.obs['Age'] = "NotAvailable"
GLIO5_T.obs['Sorted'] = "CD45"

In [18]:
GLIO6_T.obs['SampleID'] = "GLIO6_T"
GLIO6_T.obs['GEO_RNA'] = "NotAvailable"
GLIO6_T.obs['Cohort'] = "NotAvailable"
GLIO6_T.obs['Type'] = "PrimaryTumor"  
GLIO6_T.obs['Subtype'] = "NotAvailable"
GLIO6_T.obs['Diagnosis'] = "Glioblastoma"
GLIO6_T.obs['Grade'] = "NotAvailable"
GLIO6_T.obs['Location'] = "NotAvailable"
GLIO6_T.obs['IDH1_Status'] = "WT"
GLIO6_T.obs['EGFR_Status'] = "NotAvailable"
GLIO6_T.obs['MGMT'] = "UnMet"
GLIO6_T.obs['CellCount'] = "3014"
GLIO6_T.obs['Sample'] = "Patient34"
GLIO6_T.obs['Sex'] = "M"
GLIO6_T.obs['Age'] = "NotAvailable"
GLIO6_T.obs['Sorted'] = "CD45"

GLIO7_H.obs['SampleID'] = "GLIO7_H"
GLIO7_H.obs['GEO_RNA'] = "NotAvailable"
GLIO7_H.obs['Cohort'] = "NotAvailable"
GLIO7_H.obs['Type'] = "Normal"  
GLIO7_H.obs['Subtype'] = "NotAvailable"
GLIO7_H.obs['Diagnosis'] = "Astrocytoma"
GLIO7_H.obs['Grade'] = "NotAvailable"
GLIO7_H.obs['Location'] = "NotAvailable"
GLIO7_H.obs['IDH1_Status'] = "WT"
GLIO7_H.obs['EGFR_Status'] = "NotAvailable"
GLIO7_H.obs['MGMT'] = "UnMet"
GLIO7_H.obs['CellCount'] = "4032"
GLIO7_H.obs['Sample'] = "Patient35"
GLIO7_H.obs['Sex'] = "F"
GLIO7_H.obs['Age'] = "NotAvailable"
GLIO7_H.obs['Sorted'] = "CD45"

In [19]:
GLIO7_T.obs['SampleID'] = "GLIO7_T"
GLIO7_T.obs['GEO_RNA'] = "NotAvailable"
GLIO7_T.obs['Cohort'] = "NotAvailable"
GLIO7_T.obs['Type'] = "PrimaryTumor"  
GLIO7_T.obs['Subtype'] = "NotAvailable"
GLIO7_T.obs['Diagnosis'] = "Astrocytoma"
GLIO7_T.obs['Grade'] = "NotAvailable"
GLIO7_T.obs['Location'] = "NotAvailable"
GLIO7_T.obs['IDH1_Status'] = "WT"
GLIO7_T.obs['EGFR_Status'] = "NotAvailable"
GLIO7_T.obs['MGMT'] = "UnMet"
GLIO7_T.obs['CellCount'] = "4530"
GLIO7_T.obs['Sample'] = "Patient35"
GLIO7_T.obs['Sex'] = "F"
GLIO7_T.obs['Age'] = "NotAvailable"
GLIO7_T.obs['Sorted'] = "CD45"

GLIO8_H.obs['SampleID'] = "GLIO8_H"
GLIO8_H.obs['GEO_RNA'] = "NotAvailable"
GLIO8_H.obs['Cohort'] = "NotAvailable"
GLIO8_H.obs['Type'] = "PrimaryTumor"  
GLIO8_H.obs['Subtype'] = "NotAvailable"
GLIO8_H.obs['Diagnosis'] = "NotAvailable"
GLIO8_H.obs['Grade'] = "NotAvailable"
GLIO8_H.obs['Location'] = "NotAvailable"
GLIO8_H.obs['IDH1_Status'] = "WT"
GLIO8_H.obs['EGFR_Status'] = "NotAvailable"
GLIO8_H.obs['MGMT'] = "UnMet"
GLIO8_H.obs['CellCount'] = "6401"
GLIO8_H.obs['Sample'] = "Patient35"
GLIO8_H.obs['Sex'] = "F"
GLIO8_H.obs['Age'] = "NotAvailable"
GLIO8_H.obs['Sorted'] = "CD45"

In [20]:
%who

GLIO1_T	 GLIO2_T	 GLIO3_T	 GLIO4_T	 GLIO5_T	 GLIO6_T	 GLIO7_H	 GLIO7_T	 GLIO8_H	 
NamespaceMagics	 get_ipython	 getsizeof	 h5ad_create	 isfile	 join	 json	 listdir	 np	 
onlydir	 outputPath	 pd	 sc	 var_dic_list	 


In [21]:
GLIO1_T.obs = GLIO1_T.obs.reset_index(drop=True)
GLIO2_T.obs = GLIO2_T.obs.reset_index(drop=True)
GLIO3_T.obs = GLIO3_T.obs.reset_index(drop=True)
GLIO4_T.obs = GLIO4_T.obs.reset_index(drop=True)
GLIO5_T.obs = GLIO5_T.obs.reset_index(drop=True)
GLIO6_T.obs = GLIO6_T.obs.reset_index(drop=True)
GLIO7_H.obs = GLIO7_H.obs.reset_index(drop=True)
GLIO7_T.obs = GLIO7_T.obs.reset_index(drop=True)
GLIO8_H.obs = GLIO8_H.obs.reset_index(drop=True)

In [22]:
GLIO1_T.obs.set_index("GLIO1_T-" + GLIO1_T.obs.index.astype(str),inplace=True)
GLIO2_T.obs.set_index("GLIO2_T-" + GLIO2_T.obs.index.astype(str),inplace=True)
GLIO3_T.obs.set_index("GLIO3_T-" + GLIO3_T.obs.index.astype(str),inplace=True)
GLIO4_T.obs.set_index("GLIO4_T-" + GLIO4_T.obs.index.astype(str),inplace=True)
GLIO5_T.obs.set_index("GLIO5_T-" + GLIO5_T.obs.index.astype(str),inplace=True)
GLIO6_T.obs.set_index("GLIO6_T-" + GLIO6_T.obs.index.astype(str),inplace=True)
GLIO7_H.obs.set_index("GLIO7_H-" + GLIO7_H.obs.index.astype(str),inplace=True)
GLIO7_T.obs.set_index("GLIO7_T-" + GLIO7_T.obs.index.astype(str),inplace=True)
GLIO8_H.obs.set_index("GLIO8_H-" + GLIO8_H.obs.index.astype(str),inplace=True)

In [23]:
GLIO1_T.var_names_make_unique()
GLIO2_T.var_names_make_unique()
GLIO3_T.var_names_make_unique()
GLIO4_T.var_names_make_unique()
GLIO5_T.var_names_make_unique()
GLIO6_T.var_names_make_unique()
GLIO7_H.var_names_make_unique()
GLIO7_T.var_names_make_unique()
GLIO8_H.var_names_make_unique()

In [24]:
GLIO1_T.write(outputPath+"GLIO1_T.h5ad")
GLIO2_T.write(outputPath+"GLIO2_T.h5ad")
GLIO3_T.write(outputPath+"GLIO3_T.h5ad")
GLIO4_T.write(outputPath+"GLIO4_T.h5ad")
GLIO5_T.write(outputPath+"GLIO5_T.h5ad")
GLIO6_T.write(outputPath+"GLIO6_T.h5ad")
#GLIO7_H.write(outputPath+"GLIO7_H.h5ad")
GLIO7_T.write(outputPath+"GLIO7_T.h5ad")
#GLIO8_H.write(outputPath+"GLIO8_H.h5ad")